server/events/project_locker.go

```diff
@@ -58,8 +59,13 @@ type TryLockResponse struct {
 }
 
 // TryLock implements ProjectLocker.TryLock.
-func (p *DefaultProjectLocker) TryLock(log logging.SimpleLogging, pull models.PullRequest, user models.User, workspace string, project models.Project) (*TryLockResponse, error) {
-	lockAttempt, err := p.Locker.TryLock(project, workspace, pull, user)
+func (p *DefaultProjectLocker) TryLock(log logging.SimpleLogging, pull models.PullRequest, user models.User, workspace string, project models.Project, disableRepoLocking bool) (*TryLockResponse, error) {
+	locker := p.Locker
+	if disableRepoLocking {
+		locker = p.NoOpLocker
+	}
+
+	lockAttempt, err := locker.TryLock(project, workspace, pull, user)
```

Comment:
This is the main logic in this PR. just check `disableRepoLocking` and switch the locker into `NoOpLocker`.